neuralmonkey/config/utils.py

```diff
@@ -30,13 +33,13 @@ def dep_func(*args, **kwargs):
 initialize_vocabulary = vocabulary.initialize_vocabulary
 
 
-def adam_optimizer(learning_rate=1e-4):
+def adam_optimizer(learning_rate=1e-4) -> tf.train.AdamOptimizer:
     return tf.train.AdamOptimizer(learning_rate)
 
 
-def adadelta_optimizer(**kwargs):
+def adadelta_optimizer(**kwargs) -> tf.train.AdadeltaOptimizer:
     return tf.train.AdadeltaOptimizer(**kwargs)
 
 
-def variable(initial_value=0, trainable=False, **kwargs):
+def variable(initial_value=0, trainable=False, **kwargs) -> tf.Variable:
```

Comment:
Ad „tak to přidá do toho sjednocení“: to je bruální porušení všech principů typového systému. Ta anotace je kontrakt, který se autor té funkce zavázal dodržet. Tj. `a: str, b: int -> str` znamená: „dej mi string a celé číslo a já vrátím string“. Přece nemůžu předpokládat, že si uživatel té funkce bude ten kontrakt svévolně měnit. To pak ty typy postrádají veškerý smysl (ačkoliv v případě Pythonu by se hodilo spíš říct „i poslední zbytek smyslu“).